app/components/image_attachment_table_component.html.erb

```diff
@@ -0,0 +1,48 @@
+<div class="card">
+  <!-- table -->
+  <div class="table-responsive overflow-y-hidden">
+    <table class="table mb-0 text-nowrap table-hover table-centered">
+      <thead>
+        <tr>
+          <th scope="col">Image</th>
+          <th scope="col">Filename</th>
+          <th scope="col"></th>
+        </tr>
+      </thead>
+      <tbody>
+        <% if images.any?  %>
+          <% images.each do |image| %>
+            <% if image.record.persisted? %>
+              <tr>
+                <td>
+                  <div class="d-flex align-items-center">
+                    <div class="icon-shape img-4by3-lg rounded-3">
+                      <%= link_to(image_tag(image, class: "card-img"), rails_blob_path(image, disposition: "attachment")) %>
+                    </div>
+                  </div>
+                </td>
+                <td>
+                  <%= link_to image.filename.to_s, rails_blob_path(image, disposition: "attachment") %>
+                </td>
+                <td>
+                  <%= link_to t("general.delete"),
+                        staff_purge_attachment_path(image.id),
+                        data: {
+                          turbo_confirm: "Do you want to delete this image?",
+                          turbo_method: "delete" },
+                          class: "btn btn-outline-danger" %>
+                </td>
+              </tr>
+            <% end %>
+          <% end %>
+        <% else %>
+          <tr>
+            <td colspan="3">
+              No images attached.
+            </td>
+          </tr>
```

Comment:
One thing I've done is to have a general component that wraps collection lists like this.

That component takes a "blank" block, and it's render looks something like:
```ruby
def render
  collection.empty?  ? blank : content
end
```

and then the main block content can handle the rendering of everything as though it's Completely Normal, without running conditionals everywhere.

Reply:
I like the premise of this abstraction, but I think I am missing the implementation details.

I could do something like this:
```ruby
# app/components/collection_component.rb

class CollectionComponent < ApplicationComponent
  param :collection

  renders_one :blank

  def call
    collection.empty? ? blank : content
  end
end
```

```erb
<%# app/components/image_attachment_table_component.html.erb %>

  <%= render CollectionComponent.new(images) do |c| %>
    <% images.each do |image| %>
      <%# render image content %>
    <% end %>
    <%= c.with_blank do %>
      <%# blank content %>
    <% end %>
  <% end %>
```

But this feels like it's just changing syntax from an if...else block to passing two blocks to an abstract component. It also isn't actually using the collection param in the template. Can you help me clean this up?

Reply:
I am gonna go ahead and get this merged and make a note for this abstraction on the projects tab, so that we can consider refactoring this later.